<a href="https://colab.research.google.com/github/Pradip240/IPL-T20-Cricket-Analysis/blob/main/Ameen_IPL_T20_Cricket_Analysis_Capstone_Project.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

## <b> Indian Premier League(IPL) is a professional Twenty20 cricket league in India contested during March or April and May of every year by eight teams representing eight different cities in India. The league was founded by the Board of Control for Cricket in India(BCCI) in 2008. </b>

## <b> Explore and analyze the data to discover results and statistics for different teams playing in IPL. </b>

##Let's load and explore the dataset

In [2]:
from google.colab import drive
drive.mount('/content/drive')

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


In [3]:
#Importing the Numpy & the Pandas libraries.
import numpy as np
import pandas as pd

In [4]:
file_path = '/content/drive/MyDrive/AlmaBetter Data Science/1. PYTHON/CAPSTONE PROJECT/IPL Data/'
ipl_df = pd.read_csv(file_path + 'matches.csv')

In [5]:
#Checking the shape of a dataframe
ipl_df.shape


(756, 18)

In [30]:
#Checking the first five observations
ipl_df.head()

id    Season  ...         umpire1        umpire2
0   1  IPL-2017  ...     AY Dandekar       NJ Llong
1   2  IPL-2017  ...  A Nand Kishore         S Ravi
2   3  IPL-2017  ...     Nitin Menon      CK Nandan
3   4  IPL-2017  ...    AK Chaudhary  C Shamshuddin
4   5  IPL-2017  ...             NaN            NaN

[5 rows x 17 columns]

In [ ]:
#Checking the last five observations
ipl_df.tail()

In [ ]:
#Checking for the null values in the dataset
ipl_df.isnull()

In [ ]:
#Checking the info of the datset
ipl_df.info()


In [ ]:
#suming up together the null values for each column
ipl_df.isnull().sum()

In [28]:
#Looking up for the columns in the dataset
ipl_df.columns

Index(['id', 'Season', 'city', 'date', 'team1', 'team2', 'toss_winner',
       'toss_decision', 'result', 'dl_applied', 'winner', 'win_by_runs',
       'win_by_wickets', 'player_of_match', 'venue', 'umpire1', 'umpire2'],
      dtype='object')

In [ ]:
#Getting the unique dates.
ipl_df['date'].unique()

# So what can we expect from the dataset?
##In order to understand our data, we can look at each variable and try to understand their meaning and relevance to this problem.
##The datset consists of 18 columns. Let's go ahead & get familiar with the columns.
##*id: Basically the IPL match ID.
##*Season: The IPL season, which in our dataset if from season 2008 - season 2019
##*City: The cities where the IPL matches were played.
##*Date: The date on which the matches were played or held.
##*Team1 & Team2: The teams participated in the IPL matches since 2008.
##*Toss_winner: The team that won the toss.
##*Toss_decision: The decision taken by the toss winning team either to 'Bat' or 'Field' first.
##*Result: The result column consists of (normal, tie, no result) of the match.
##Dl_applied: The column consists the numerical values (0 or 1) which indicates whether the Duckworth-Lewis method was applied or not.
##*Winner: Winner of the respective match.
##*Win_by_runs: The runs by which the team batting first won.
##*Win_by_wickets: The wickets by which the team batting second won.
##*Player_of_the_match: The player who performed exceptionally good in the match.
##*Venue: The venue where the match was hosted.
##*Umpire1 & Umpire2: Two on-field umpires during the match.
##*Umpire3: Basically the third Umpire.

##Observations from the dataset:
##1.The dataset has 756 rows and 18 columns.
##2.The dataset hardly has any null values apart from column 'umpire3' which has significant null values. (i.e 637 null values)
##3.So we can drop the 'umpire3' column from the dataframe and proceed further with the EDA. 

In [26]:
#Droping the 'Umpire3' column from the dataset
ipl_df = ipl_df.drop(columns=['umpire3'], axis=1)

In [ ]:
ipl_df.head()

##As our dataset has been cleaned, we can now do the in-depth analysis and visualization.